pkg/provider/response.go

```diff
@@ -12,17 +14,17 @@ import (
 	"github.com/zitadel/saml/pkg/provider/xml/samlp"
 )
 
-const (
-	StatusCodeSuccess                = "urn:oasis:names:tc:SAML:2.0:status:Success"
-	StatusCodeVersionMissmatch       = "urn:oasis:names:tc:SAML:2.0:status:VersionMismatch"
-	StatusCodeAuthNFailed            = "urn:oasis:names:tc:SAML:2.0:status:AuthnFailed"
-	StatusCodeInvalidAttrNameOrValue = "urn:oasis:names:tc:SAML:2.0:status:InvalidAttrNameOrValue"
-	StatusCodeInvalidNameIDPolicy    = "urn:oasis:names:tc:SAML:2.0:status:InvalidNameIDPolicy"
-	StatusCodeRequestDenied          = "urn:oasis:names:tc:SAML:2.0:status:RequestDenied"
-	StatusCodeRequestUnsupported     = "urn:oasis:names:tc:SAML:2.0:status:RequestUnsupported"
-	StatusCodeUnsupportedBinding     = "urn:oasis:names:tc:SAML:2.0:status:UnsupportedBinding"
-	StatusCodeResponder              = "urn:oasis:names:tc:SAML:2.0:status:Responder"
-	StatusCodePartialLogout          = "urn:oasis:names:tc:SAML:2.0:status:PartialLogout"
+var (
+	statusCodeSuccess                = errors.New("urn:oasis:names:tc:SAML:2.0:status:Success")
+	StatusCodeVersionMissmatch       = errors.New("urn:oasis:names:tc:SAML:2.0:status:VersionMismatch")
+	StatusCodeAuthNFailed            = errors.New("urn:oasis:names:tc:SAML:2.0:status:AuthnFailed")
+	StatusCodeInvalidAttrNameOrValue = errors.New("urn:oasis:names:tc:SAML:2.0:status:InvalidAttrNameOrValue")
+	StatusCodeInvalidNameIDPolicy    = errors.New("urn:oasis:names:tc:SAML:2.0:status:InvalidNameIDPolicy")
+	StatusCodeRequestDenied          = errors.New("urn:oasis:names:tc:SAML:2.0:status:RequestDenied")
+	StatusCodeRequestUnsupported     = errors.New("urn:oasis:names:tc:SAML:2.0:status:RequestUnsupported")
+	StatusCodeUnsupportedBinding     = errors.New("urn:oasis:names:tc:SAML:2.0:status:UnsupportedBinding")
+	StatusCodeResponder              = errors.New("urn:oasis:names:tc:SAML:2.0:status:Responder")
+	StatusCodePartialLogout          = errors.New("urn:oasis:names:tc:SAML:2.0:status:PartialLogout")
```

Comment:
why is this (breaking) change needed? they're anyway just used as error.Error()

Reply:
also making success an error and private?

Reply:
Rollbacked the changes here.